packages/zwave-js/src/lib/commandclass/UserCodeCC.ts

```diff
@@ -236,6 +236,48 @@ function validateCode(code: string, supportedChars: string): boolean {
 	return [...code].every((char) => supportedChars.includes(char));
 }
 
+function setUserCodeMetadata(this: UserCodeCC, userId: number) {
```

Comment:
@AlCalzone I've moved the setting of the user code metadata to a separate function because it appears we need to call this functionality in a couple of different places. I've also gone ahead and **assumed** that the default type for a code is a string. Let me know if this is ok.

Actually quick question: do we want to be able to override the metadata type for the user code to Buffer in the event that it is in fact a buffer?

Reply:
> Actually quick question: do we want to be able to override the metadata type for the user code to Buffer in the event that it is in fact a buffer?

Yeah, we need to be able to do that, when a device responds with a user code that needs to be represented as a buffer. I'd add an optional `userCode` parameter here and use that to distinguish. The current code will always set the metadata to string.